### ISSUE TITLE
git tree is significantly dirtied after running test suite

### ISSUE THREAD

**Issue by RyanGlScott** [haskell/haddock#1030]
After running the Haddock tests through GHC's test suite, the `haddock` tree is left cluttered with several files that dirty the `git status`:

```
$ git status
On branch visible-dependent-quantification-take-3
Untracked files:
  (use "git add <file>..." to include in what will be committed)

        haddock-test/src/Test/Haddock.hi
        haddock-test/src/Test/Haddock/Config.hi
        haddock-test/src/Test/Haddock/Process.hi
        haddock-test/src/Test/Haddock/Utils.hi
        haddock-test/src/Test/Haddock/Xhtml.hi
        hoogle-test/Main.hi
        html-test/Main.hi
        hypsrc-test/Main.hi
        latex-test/Main.hi
```

Can these be added to `.gitignore`?

**Comment by harpocrates** [comment]
I forget what the `ghc` option is to say: "don't write out any interface files!", but that should also be added to `haddock(Html|Hoogle|Latex|Hypsrc)Test`.

**Comment by RyanGlScott** [comment]
That would be a nice thing to add. But is there any reason not to also add `*.hi` to Haddock's `.gitignore` as well? (This is what [GitHub's standard `.gitignore` template for Haskell](https://github.com/github/gitignore/blob/d2c1bb2b9c72ead618c9f6a48280ebc7a8e0dff6/Haskell.gitignore) does, as one data point in favor of this.)

**Comment by harpocrates** [comment]
Oh no, I'm in support of that too, hence my thumbs up! TBH, I'm surprised this isn't something we don't already have.

**Comment by harpocrates** [comment]
Fixed in 50cf001765123a6c787795dda50aaefa5f2beed7 and https://github.com/ghc/ghc/commit/e2ae52c34e81123dce9991fad393d96fd7b7af6a.
